sklearn/feature_extraction/text.py

```diff
@@ -242,6 +242,8 @@ def build_analyzer(self):
     def _validate_vocabulary(self):
         vocabulary = self.vocabulary
         if vocabulary is not None:
+            if isinstance(vocabulary, set):
+                self.vocabulary = list(vocabulary)
```

Comment:
Unfortunately we probably shouldn't change self.vocabulary, which should only be set in ``__init__`` Does it still work if we only do this locally here?